src/VisualStudio/Core/Def/Implementation/LanguageClient/InProcLanguageServer.cs

```diff
@@ -90,79 +89,79 @@ public void Exit()
         {
-            var documentFormattingParams = input.ToObject<DocumentFormattingParams>();
+            var documentFormattingParams = input.ToObject<DocumentFormattingParams>(JsonSerializer);
             return _protocol.FormatDocumentAsync(_workspace.CurrentSolution, documentFormattingParams, _clientCapabilities, cancellationToken);
         }
 
         [JsonRpcMethod(Methods.TextDocumentOnTypeFormattingName)]
         public Task<TextEdit[]> GetTextDocumentFormattingOnTypeAsync(JToken input, CancellationToken cancellationToken)
         {
-            var documentOnTypeFormattingParams = input.ToObject<DocumentOnTypeFormattingParams>();
+            var documentOnTypeFormattingParams = input.ToObject<DocumentOnTypeFormattingParams>(JsonSerializer);
             return _protocol.FormatDocumentOnTypeAsync(_workspace.CurrentSolution, documentOnTypeFormattingParams, _clientCapabilities, cancellationToken);
         }
 
         [JsonRpcMethod(Methods.TextDocumentImplementationName)]
-        public Task<object> GetTextDocumentImplementationsAsync(JToken input, CancellationToken cancellationToken)
+        public Task<SumType<Location, Location[]>?> GetTextDocumentImplementationsAsync(JToken input, CancellationToken cancellationToken)
         {
-            var textDocumentPositionParams = input.ToObject<TextDocumentPositionParams>();
+            var textDocumentPositionParams = input.ToObject<TextDocumentPositionParams>(JsonSerializer);
             return _protocol.FindImplementationsAsync(_workspace.CurrentSolution, textDocumentPositionParams, _clientCapabilities, cancellationToken);
         }
 
         [JsonRpcMethod(Methods.TextDocumentRangeFormattingName)]
         public Task<TextEdit[]> GetTextDocumentRangeFormattingAsync(JToken input, CancellationToken cancellationToken)
         {
-            var documentRangeFormattingParams = input.ToObject<DocumentRangeFormattingParams>();
+            var documentRangeFormattingParams = input.ToObject<DocumentRangeFormattingParams>(JsonSerializer);
             return _protocol.FormatDocumentRangeAsync(_workspace.CurrentSolution, documentRangeFormattingParams, _clientCapabilities, cancellationToken);
         }
 
         [JsonRpcMethod(Methods.TextDocumentSignatureHelpName)]
         public async Task<SignatureHelp> GetTextDocumentSignatureHelpAsync(JToken input, CancellationToken cancellationToken)
         {
-            var textDocumentPositionParams = input.ToObject<TextDocumentPositionParams>();
+            var textDocumentPositionParams = input.ToObject<TextDocumentPositionParams>(JsonSerializer);
             return await _protocol.GetSignatureHelpAsync(_workspace.CurrentSolution, textDocumentPositionParams, _clientCapabilities, cancellationToken).ConfigureAwait(false);
         }
 
         [JsonRpcMethod(Methods.WorkspaceSymbolName)]
         public async Task<SymbolInformation[]> GetWorkspaceSymbolsAsync(JToken input, CancellationToken cancellationToken)
```

Comment:
I think you can omit the `JToken` and still be backwards compatible if you use the right signature and version of StreamJsonRpc.

```
        [JsonRpcMethod(Methods.WorkspaceSymbolName, UseSingleObjectParameterDeserialization = true)]
        public async Task<SymbolInformation[]> WorkspaceSymbolAsync(
            WorkspaceSymbolParams args,
            CancellationToken cancellationToken)
        {
            ..
        }
```
https://github.com/dotnet/roslyn/issues/41115